uiowa/uiowa
# Create articles, people and events Layout Builder blocks.

Resolves #1789, resolves #2071 

Introduces three new custom block types (articles, people, events) that are basically ports of the paragraph versions.

Created https://github.com/uiowa/uiowa/issues/2058 to address alternative layout/style options

## To Do

- [x] Fix caching issues with collection heading
- [x] Remove border-top, consistent top margin
- [x] Recreate and share fields between articles/people and events block
- [x] Add collection heading to events block
- [x] Add LB styles to events block
- [x] Add event location to teaser and style appropriately for hide-details LB style.
- [ ] ~~Update event styles to match existing in UIDS.~~
- [x] Update sitenow_events instance date formats.

## To Test

- Using a v3 site with articles, people and events (more than one of each), pull/sync. 
  - The default site can reinstalled locally as v3 by running `blt drupal:install`.
- Compile frontend SASS via ` blt frontend` or `yarn workspace uids_base gulp`.
- As a webmaster ensure there are three new blocks available (Articles, People, Events). Configure/place as you like.
- Within the layout interface and the rendered content (depending on content available) you should see a list of content and your configuration should match the output (hiding images, displaying a certain number, displaying a pager if more than number of items, filtered by tag, more link with custom path, etc.).
- Set heading field and make sure the teaser markup includes the correct heading tag (not the h5 class).
- Out of scope: there is intentionally no grid, masonry or other layout options included with this PR.
- For events, test out a single event page by changing the link path config at Admin -> Events. (drush cr afterwards).

I have not built this with collegiate split in mind as I am sure there are at least permission updates involved.
